bool joint_1
bool joint_2
bool joint_3
bool joint_4
bool joint_5
bool joint_6